File: frontend/src/lib/api/types/workflow.ts
export type WorkflowMode = "full" | "resume" | "splitwise_only";

export type WorkflowJobStatus =
  | "pending"
  | "running"
  | "completed"
  | "failed"
  | "cancelled";

export type WorkflowEventLevel = "info" | "success" | "warning" | "error";

export interface WorkflowEvent {
  event: string;
  step: string;
  message: string;
  level: WorkflowEventLevel;
  account: string | null;
  timestamp: string;
  data: Record<string, unknown>;
}

export interface WorkflowRunRequest {
  mode: WorkflowMode;
  start_date?: string | null;
  end_date?: string | null;
  splitwise_start_date?: string | null;
  splitwise_end_date?: string | null;
  enable_secondary_account?: boolean | null;
  override?: boolean;
}

export interface WorkflowRunResponse {
  job_id: string;
  mode: WorkflowMode;
  started_at: string;
}

export interface WorkflowJobStatusResponse {
  job_id: string;
  status: WorkflowJobStatus;
  mode: WorkflowMode;
  started_at: string;
  completed_at: string | null;
  events: WorkflowEvent[];
  summary: Record<string, unknown> | null;
  error: string | null;
}

export interface WorkflowCancelResponse {
  job_id: string;
  status: "cancelling";
}

export interface WorkflowPeriodCheckIncomplete {
  normalized_filename: string;
  status: string;
  account_nickname: string | null;
}

export interface WorkflowPeriodCheck {
  month: string;
  total: number;
  complete: number;
  incomplete: WorkflowPeriodCheckIncomplete[];
  all_done: boolean;
  has_partial: boolean;
}

File: backend/src/apis/schemas/workflow.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class WorkflowMode(str, Enum):
    full = "full"
    resume = "resume"
    splitwise_only = "splitwise_only"


class WorkflowEventLevel(str, Enum):
    info = "info"
    success = "success"
    warning = "warning"
    error = "error"


class WorkflowEvent(BaseModel):
    event: str
    step: str
    message: str
    level: WorkflowEventLevel = WorkflowEventLevel.info
    account: Optional[str] = None
    timestamp: datetime
    data: Dict[str, Any] = {}


class WorkflowRunRequest(BaseModel):
    mode: WorkflowMode = WorkflowMode.full
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    splitwise_start_date: Optional[str] = None
    splitwise_end_date: Optional[str] = None
    enable_secondary_account: Optional[bool] = None
    override: bool = False


class WorkflowRunResponse(BaseModel):
    job_id: str
    mode: WorkflowMode
    started_at: datetime


class WorkflowJobStatus(str, Enum):
    pending = "pending"
    running = "running"
    completed = "completed"
    failed = "failed"
    cancelled = "cancelled"


class WorkflowJobStatusResponse(BaseModel):
    job_id: str
    status: WorkflowJobStatus
    mode: WorkflowMode
    started_at: datetime
    completed_at: Optional[datetime] = None
    events: List[WorkflowEvent] = []
    summary: Optional[Dict[str, Any]] = None
    error: Optional[str] = None
